integrations/llamaindex/tests/test_toolkit.py

```diff
@@ -0,0 +1,162 @@
+
+
+class FunctionObj(NamedTuple):
+    full_function_name: str
+    comment: str
+
+
+@contextmanager
+def create_function_and_cleanup(client: DatabricksFunctionClient):
+    func_name = f"{CATALOG}.{SCHEMA}.test_{uuid.uuid4().hex[:4]}"
+    comment = "Executes Python code and returns its stdout."
+    sql_body = f"""CREATE OR REPLACE FUNCTION {func_name}(code STRING COMMENT 'Python code to execute. Remember to print the final result to stdout.')
+RETURNS STRING
+LANGUAGE PYTHON
+COMMENT '{comment}'
+AS $$
+    import sys
+    from io import StringIO
+    stdout = StringIO()
+    sys.stdout = stdout
+    exec(code)
+    return stdout.getvalue()
+$$
+"""
+    try:
+        client.create_function(sql_function_body=sql_body)
+        yield FunctionObj(full_function_name=func_name, comment=comment)
+    finally:
+        try:
+            client.client.functions.delete(func_name)
+        except Exception as e:
+            _logger.warning(f"Failed to delete function: {e}")
+
+
+@pytest.fixture
+def client():
+    with mock.patch(
+        "unitycatalog.ai.databricks.get_default_databricks_workspace_client",
+        return_value=mock.Mock(),
+    ):
+        yield DatabricksFunctionClient(warehouse_id="warehouse_id", cluster_id="cluster_id")
+
+
+@pytest.fixture
+def set_default_client(client):
+    set_uc_function_client(client)
+    yield
+    set_uc_function_client(None)
+
+
+@requires_databricks
+def test_toolkit_e2e(set_default_client):
+    client = get_uc_function_client()
+    with create_function_and_cleanup(client) as func_obj:
+        toolkit = LlamaIndexToolkit(function_names=[func_obj.full_function_name])
+        tools = toolkit.tools
+        assert len(tools) == 1
+        tool = tools[0]
+        assert tool.metadata.name == func_obj.full_function_name
+        assert tool.metadata.description == func_obj.comment
+        assert tool.client_config == client.to_dict()
+
+        input_args = {"code": "print(1)"}
+        result = json.loads(tool.fn(**input_args))["value"]
+        assert result == "1\n"
+
+        # Test wildcard function names
+        toolkit = LlamaIndexToolkit(function_names=[f"{CATALOG}.{SCHEMA}.*"])
+        assert len(toolkit.tools) >= 1
+        assert func_obj.full_function_name in [t.metadata.name for t in toolkit.tools]
+
+
+@requires_databricks
+def test_multiple_toolkits(set_default_client):
+    client = get_uc_function_client()
+    with create_function_and_cleanup(client) as func_obj:
+        toolkit1 = LlamaIndexToolkit(function_names=[func_obj.full_function_name])
+        toolkit2 = LlamaIndexToolkit(function_names=[f"{CATALOG}.{SCHEMA}.*"])
+        tool1 = toolkit1.tools[0]
+        tool2 = [t for t in toolkit2.tools if t.metadata.name == func_obj.full_function_name][0]
+        input_args = {"code": "print(1)"}
+        result1 = json.loads(tool1.fn(**input_args))["value"]
+        result2 = json.loads(tool2.fn(**input_args))["value"]
+        assert result1 == result2
+
```

Comment:
Let's add a test that manually pass client to the LlamaIndexToolkit in case there're any bugs :)

Reply:
good idea :)